### PR TITLE
Reduce external dependencies for futures-related things

### LINKED ISSUE #251: Consider replacing futures dependency with sub-crates
glib currently depends on `futures`, pulling in all of [`futures-channel`, `future-core`, `futures-executor`, `futures-io`, `futures-sink`, `futures-task` and `futures-tokio`] as transitive dependencies. Unless you actually use functionality from all of those crates, it would be good to only depend on the specific sub-crates whose functionality is being used.

### PR DESCRIPTION
**Pull request by jplatte**
Closes #251.

Sorry for the formatting change in `gio/Cargo.toml`, I can undo that if you want.

This touches some unsafe code, but I'm fairly certain it is no less sound than before 😅

### CODE REVIEW

**Review by GuillaumeGomez — APPROVED** [review]
Less dependencies makes me happy.

**Comment by GuillaumeGomez** [comment]
The regen check failure is weird though.

**Comment by sdroege** [comment]
> The regen check failure is weird though.

Well it's because of the ` Remove unused imports ` commit

File: graphene/sys/tests/abi.rs [hunk 1]
```diff
@@ -2,10 +2,8 @@
 // from gir-files (https://github.com/gtk-rs/gir-files)
 // DO NOT EDIT
 
-use graphene_sys::*;
 use std::env;
 use std::error::Error;
-use std::mem::{align_of, size_of};
```
Comment (sdroege) [inline graphene/sys/tests/abi.rs:8]: This breaks the CI

File: graphene/sys/tests/abi.rs [hunk 2]
```diff
@@ -2,10 +2,8 @@
 // from gir-files (https://github.com/gtk-rs/gir-files)
 // DO NOT EDIT
 
-use graphene_sys::*;
```
Comment (bilelmoussaoui) [inline graphene/sys/tests/abi.rs:5]: Those changes are not intended I guess?

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise

File: graphene/sys/tests/abi.rs (hunk 1 above)
Comment (jplatte) [reply graphene/sys/tests/abi.rs:8]: That is kind of weird, IMHO there should at least be an `#[allow(unused_imports)]` then.

**Comment by jplatte** [comment]
Removed the problematic commit.

**Review by sdroege — APPROVED** [review]

**Pull request merged**